pkg/util/docker/docker_util.go

```diff
@@ -207,6 +207,60 @@ func (d *DockerUtil) ResolveImageName(image string) (string, error) {
 	return d.imageNameBySha[image], nil
 }
 
+// ResolveImageNameFromContainer will resolve the container sha image name to their user-friendly name.
+// It is similar to ResolveImageName except it tries to match the image to the container Config.Image.
+// For non-sha names we will just return the name as-is.
+func (d *DockerUtil) ResolveImageNameFromContainer(co types.ContainerJSON) (string, error) {
+	image := co.Image
+	if !strings.Contains(image, "sha256:") {
+		return image, nil
+	}
+
+	d.Lock()
+	defer d.Unlock()
```

Comment:
It looks like we are holding this lock for this entire method. Would it make sense to restructure this code a bit so that we only get/set in `d.imageNameBySha` when this lock is held? Concerned by the `ImageInspectWithRaw` within this scope.

Reply:
OK will think about this

Reply:
We could refactor this later if you'd like?